src/cpu/rv64/rvv_gemm_convolution.cpp

```diff
@@ -0,0 +1,497 @@
+                            vfloat32m1_t v_d
+                                    = __riscv_vle32_v_f32m1(d_ + oS, vl);
+                            v_d = __riscv_vfadd_vf_f32m1(v_d, b, vl);
+                            __riscv_vse32_v_f32m1(d_ + oS, v_d, vl);
+                            oS += vl;
+                        }
+                    });
+                }
+            }
+
+            return status::success;
+        };
+        im_pos_t start, end;
+        end.ic = jcp.ic;
+
+        if (!is_problem_3d) {
+            dim_t sp_work = jcp.mb * jcp.ngroups * jcp.od * jcp.os;
+            balance2D(nthr, ithr, sp_work, start.sp, end.sp, jcp.oc, start.oc,
+                    end.oc, dim_t(jcp.nthr_oc));
+        } else {
+            dim_t sp_work = jcp.mb * jcp.ngroups * jcp.od;
+            balance2D(nthr, ithr, sp_work, start.sp, end.sp, jcp.oc, start.oc,
+                    end.oc, dim_t(jcp.nthr_oc));
+            start.sp *= jcp.os;
+            end.sp *= jcp.os;
+        }
+
+        im_pos_t curr, prev, step;
+        prev.n = prev.g = prev.od = prev.sp = prev.ic = -1;
+        step.oc = jcp.oc_block;
+        step.sp = jcp.os_block;
+        step.ic = jcp.ic_block;
+
+        if (jcp.loop_order == gemm_loop_rlb)
+            for (curr.ic = 0; curr.ic < jcp.ic; curr.ic += step.ic)
+                for (int spatial = start.sp; spatial < end.sp;
+                        spatial += step.sp) {
+                    nd_iterator_init(spatial, curr.n, jcp.mb, curr.g,
+                            jcp.ngroups, curr.od, jcp.od, curr.sp, jcp.os);
+                    for (curr.oc = start.oc; curr.oc < end.oc;
+                            curr.oc += step.oc) {
+                        status_t st_thr
+                                = inner_ker(spatial, curr, prev, step, end);
+                        if (st_thr != status::success) {
+                            st = st_thr;
+                            return;
+                        }
+                    }
+                }
+        else if (jcp.loop_order == gemm_loop_lrb)
+            for (int spatial = start.sp; spatial < end.sp; spatial += step.sp) {
+                nd_iterator_init(spatial, curr.n, jcp.mb, curr.g, jcp.ngroups,
+                        curr.od, jcp.od, curr.sp, jcp.os);
+                for (curr.ic = 0; curr.ic < jcp.ic; curr.ic += step.ic)
+                    for (curr.oc = start.oc; curr.oc < end.oc;
+                            curr.oc += step.oc) {
+                        status_t st_thr
+                                = inner_ker(spatial, curr, prev, step, end);
+                        if (st_thr != status::success) {
+                            st = st_thr;
+                            return;
+                        }
+                    }
+            }
+        else
+            st = status::unimplemented;
+    });
+
+    return st;
+}
+
+} // namespace rv64
+} // namespace cpu
+} // namespace impl
+} // namespace dnnl
```

Comment:
(nit) Pls add EoL in the end of files here and other files in this PR.

```suggestion
} // namespace dnnl

```